include/laser_filters/scan_shadow_detector.h

```diff
@@ -46,8 +46,18 @@ class ScanShadowDetector
 public:
   float min_angle_tan_, max_angle_tan_;  // Filter angle thresholds
 
-  void configure(const float min_angle, const float max_angle);
-  bool isShadow(const float r1, const float r2, const float included_angle);
+  void configure(const float min_angle, const float max_angle, const int window);
+  void prepareForInput(const float angle_increment);
+  bool isShadow(const float r1, const float r2, const int angle_index);
+
+  ~ScanShadowDetector();
+private:
+  int window_;
+  float angle_increment_;
+  float *sin_map_ = nullptr;
```

Comment:
These raw pointers worry me in terms of protecting against bugs in future patches. Why not use a `std::vector<float>` that will do the memory management for us?

Reply:
Using a vector here slows down the update loop by 25% on my machine. That's significant, but overall there is still a significant performance gain. I agree that pointers are riskier. I let it be your call if you want to switch. We have the code ready.

Reply:
I wonder why there is a performance loss at all - are you compiling in release mode? (`-DCMAKE_BUILD_TYPE=Release`?) The vector shouldn't be doing any bounds checking or anything in that case, so I don't see why it wouldn't compile down to the same machine code as the raw pointers.

Reply:
When building with release settings vectors still seem slightly slower, but only 10% which is negligible. All in all, the performance gain is more than a factor three. So the code has been changed to use vectors